pytorch_lightning/trainer/progress.py

```diff
@@ -141,22 +141,22 @@ def scheduler_steps(self) -> int:
 
 
 @dataclass
-class TrainingProgress(Progress):
-    """
-    Extends ``Progress`` with training specific attributes
-
-    Args:
-        optimization: Tracks optimization progress
-    """
-    optimization: OptimizationProgress = field(default_factory=OptimizationProgress)
+class TrainingLoopProgress(LoopProgress):
 
+    optimizer_idx: Optional[int] = field(default_factory=int)
+    epoch: Progress = field(default_factory=Progress)
+    optimizations: Optional[tuple] = field(default_factory=tuple)
 
-@dataclass
-class TrainingLoopProgress(LoopProgress):
-    epoch: TrainingProgress = field(default_factory=TrainingProgress)
+    def reset_on_batch(self) -> None:
+        if self.optimizations is not None:
+            for opt in self.optimizations:
+                opt.optimizer.current.reset()
+                opt.scheduler.current.reset()
+                opt.zero_grad.current.reset()
 
     def reset_on_epoch(self) -> None:
         # override to avoid resetting `epoch.current`
+        self.epoch.current.reset()
```

Comment:
We skip this reset on purpose (see above comment)

as `epoch.current` is meant to track "number of epochs this fit() call" (see gdoc)

Reply:
I see ! I missed this part. Rectified and updated the test to break in the future.